extension/entitystore/ec2Info.go

```diff
@@ -0,0 +1,144 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package entitystore
+
+import (
+	"context"
+	"errors"
+	"strings"
+	"time"
+
+	"go.uber.org/zap"
+
+	"github.com/aws/amazon-cloudwatch-agent/internal/ec2metadataprovider"
+	"github.com/aws/amazon-cloudwatch-agent/plugins/processors/ec2tagger"
+)
+
+const (
+	// InstanceId character maximum length is 19.
+	// See https://docs.aws.amazon.com/autoscaling/ec2/APIReference/API_Instance.html.
+	instanceIdSizeMax = 19
+
+	// AutoScalingGroup character maximum length is 255.
+	// See https://docs.aws.amazon.com/autoscaling/ec2/APIReference/API_AutoScalingGroup.html.
+	autoScalingGroupSizeMax = 255
+)
+
+type EC2Info struct {
+	InstanceID       string
+	AccountID        string
+	AutoScalingGroup string
+
+	// region is used while making call to describeTags Ec2 API for AutoScalingGroup
+	Region string
+
+	metadataProvider ec2metadataprovider.MetadataProvider
+	logger           *zap.Logger
+	done             chan struct{}
+}
+
+func (ei *EC2Info) initEc2Info() {
+	ei.logger.Debug("Initializing EC2Info")
+	if err := ei.setInstanceIDAccountID(); err != nil {
+		return
+	}
+	if err := ei.setAutoScalingGroup(); err != nil {
+		return
+	}
+	ei.logger.Debug("Finished initializing EC2Info")
+	ei.ignoreInvalidFields()
+}
+
+func (ei *EC2Info) setInstanceIDAccountID() error {
+	for {
+		metadataDoc, err := ei.metadataProvider.Get(context.Background())
+		if err != nil {
+			ei.logger.Warn("Failed to get Instance ID / Account ID through metadata provider", zap.Error(err))
+			wait := time.NewTimer(1 * time.Minute)
```

Comment:
Why doesn't this use the same backoff as setASG?

Reply:
I believe, we want this to happen very minute to not send entity for cross account .

Reply:
Non-blocking. Need confirmation with @jefchien